src/storage/hub.ts

```diff
@@ -61,8 +62,16 @@ function makeLegacyAuthToken(challengeText: string, signerKeyHex: string): strin
       && parsedChallenge[3] === 'blockstack_storage_please_sign') {
     const signer = hexStringToECPair(signerKeyHex
                                      + (signerKeyHex.length === 64 ? '01' : ''))
-    const digest = bitcoin.crypto.sha256(challengeText)
-    const signature = signer.sign(digest).toDER().toString('hex')
+    const digest = bitcoin.crypto.sha256(Buffer.from(challengeText))
+
+    const signatureBuffer = signer.sign(digest)
+    const signatureWithHash = bitcoin.script.signature.encode(
+      signatureBuffer, bitcoin.Transaction.SIGHASH_NONE)
+    
+    // We only want the DER encoding so remove the sighash version byte at the end.
```

Comment:
Note to reviewers: this has been broken in master for quite some time. Since the upgrade from bitcoinjs-lib 3.x to 4.x. 
No unit tests for this and Flow had no understanding of the bitcoinjs-lib typings. Typescript identified this bug and I pulled in the already fixed and tested code from Gaia.